src/serac/physics/solid_mechanics.hpp

```diff
@@ -958,10 +1040,18 @@ class SolidMechanics<order, dim, Parameters<parameter_space...>, std::integer_se
     // u += dot(inv(J), dot(J_elim[:, dofs], (U(t + dt) - u)[dofs]));
 
     // Update the linearized Jacobian matrix
-    auto [r, drdu] = (*residual_)(differentiate_wrt(displacement_), zero_, shape_displacement_,
-                                  *parameters_[parameter_indices].state...);
-    J_             = assemble(drdu);
-    J_e_           = bcs_.eliminateAllEssentialDofsFromMatrix(*J_);
+    if (!contact_.haveLagrangeMultipliers()) {
+      residual_with_bcs_->GetGradient(displacement_);
+    } else {
+      int               disp_size = displacement_.Size();
+      mfem::Array<int>  block_offsets({0, disp_size, disp_size + contact_.numPressureTrueDofs()});
+      mfem::BlockVector augmented_solution(block_offsets);
```

Comment:
Whenever you create a block vector, I would add some comments about the memory semantics used. E.g. This block vector contains references to the independently owned and managed displacement and pressure fields.